core/Pos/Binary/Core/Blockchain.hs

```diff
@@ -32,11 +31,11 @@ instance ( Typeable b
         blockMagic <- decode
         when (blockMagic /= getProtocolMagic protocolMagic) $
             fail $ "GenericBlockHeader failed with wrong magic: " <> show blockMagic
-        prevBlock <- decode
-        bodyProof <- decode
-        consensus <- decode
-        extra     <- decode
-        eitherToFail $ T.recreateGenericHeader prevBlock bodyProof consensus extra
+        _gbhPrevBlock <- ({-# SCC "decode_header_prev" #-} decode)
```

Comment:
Oh, alright, so this PR is not supposed to go into `develop`, sorry for criticizing the style then.